### PR TITLE
fix FP8 W8A8 MoE weights (offline + online) and add test

### PR DESCRIPTION
**Pull request by tju-sqx**
## Motivation

The previous codebase did not provide matching implementations for online and offline W8A8 FP8 quantization paths for MoE. This change implements the corresponding logic so both offline-serialized  W8A8 FP8 checkpoints and online per-token/group quantization are handled consistently.

## Modifications

Add MoE FP8 handling that mirrors the offline path used by W8A8Fp8LinearMethod.
Unify dynamic (online) quantization to use per-channel quantization (per_token_group) for MoE.
Add unit tests for Qwen / Qwen3-30B-A3B  exercising the online-MoE FP8 path.

## Accuracy Tests

Testing on Qwen3-30B-A3B with these changes shows improved and more stable accuracy.

## Checklist

- [Done] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [Done] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [N/A] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [Done] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ Done] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).
- [ ] Work with maintainers to merge your PR. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process)

### CODE REVIEW

File: python/sglang/srt/layers/quantization/w8a8_fp8.py [hunk 1]
```diff
@@ -271,14 +271,46 @@ def create_weights(
...
-        layer.w13_weight = Parameter(layer.w13_weight, requires_grad=False)
-        layer.w2_weight = Parameter(layer.w2_weight, requires_grad=False)
-        layer.w13_weight_scale = Parameter(
-            layer.w13_weight_scale.data, requires_grad=False
-        )
-        layer.w2_weight_scale = Parameter(
-            layer.w2_weight_scale.data, requires_grad=False
-        )
+        def _process_weights(weight_name: str, scale_name: str):
+            """Helper to process a single expert weight."""
```
Comment (AniZpZ) [inline python/sglang/srt/layers/quantization/w8a8_fp8.py:275]: w, n = getattr(layer, weight_name)  for once?

Comment (tju-sqx) [reply python/sglang/srt/layers/quantization/w8a8_fp8.py]: Fixed
